rally_ovs/plugins/ovs/scenarios/ovn.py

```diff
@@ -192,15 +192,18 @@ def _list_acl(self, lswitches):
 
     @atomic.action_timer("ovn.delete_acl")
     def _delete_acl(self, lswitches):
-        LOG.info("delete ACLs")
-
         ovn_nbctl = self.controller_client("ovn-nbctl")
-        ovn_nbctl.set_sandbox("controller-sandbox")
         ovn_nbctl.enable_batch_mode(True)
         for lswitch in lswitches:
-            LOG.info("delete ACLs on lswitch %s" % lswitch["name"])
-            ovn_nbctl.acl_del(lswitch["name"])
+            self._delete_single_acl(lswitch)
+        ovn_nbctl.flush()
 
+    def _delete_single_acl(self, lswitch, direction=None,
+                           priority=None, match=None):
+        ovn_nbctl = self.controller_client("ovn-nbctl")
+        ovn_nbctl.set_sandbox("controller-sandbox")
+        LOG.info("delete ACLs on lswitch %s" % lswitch["name"])
+        ovn_nbctl.acl_del(lswitch["name"], direction, priority, match)
         ovn_nbctl.flush()
```

Comment:
This flush() will result in command executed for every lswitch and makes the batch mode set in L196 useless. It should be removed.

However, I would suggest let's just update the original _delete_acl() function by adding the optional parameters, instead of adding a new function. The new function is named _delete_single_acl, but in fact this is not true, and not necessary either. When we want to delete a specific ACL, just call _delete_acl() with specific parameters.

Reply:
I agree, but if I remove the flush() in _delete_single_acl() the operation is not performed since batch mode is not properly disabled in the processing (as we saw with address_set pacthset)
Ok, agree so we can add direction, priority and match parameters to _delete_acl and pass a list composed by just one lswitch to it (in case I want to remove a given rule from a specific switch). If I use default values for _delete_acl, it will delete all the ACL rules from the swicthes in lswitches list. Agree?

Reply:
> but if I remove the flush() in _delete_single_acl() the operation is not performed

That's not true, because you already added a flush() at L199, which will make sure all commands appended in the loop finally being executed as a single transaction.

> Ok, agree so we can add direction, priority and match parameters to _delete_acl and pass a list composed by just one lswitch to it (in case I want to remove a given rule from a specific switch). If I use default values for _delete_acl, it will delete all the ACL rules from the swicthes in lswitches list. Agree?

Well, I took a second look, and I think it makes sense to have separate functions to delete all ACLs in many switches, and delete ACL(s) in a single switch. So we'd better just rename the functions: _delete_acl() => _delete_all_acls_in_lswitches(), _delete_single_acl => _delete_acls(). Would this makes the code more clear?